hal/armv7a/exceptions.c

```diff
@@ -19,6 +19,7 @@
 #include "../console.h"
 #include "../string.h"
 #include "../../include/mman.h"
+#include "../../proc/threads.h"
```

Comment:
This should be avoided, sadly we indeed need one function from `proc` module, this can be solved like here: https://github.com/phoenix-rtos/phoenix-rtos-kernel/blob/4cdff06e1309956cb835aabc4ddcde42d7587441/hal/armv7m/stm32/interrupts.c#L76